qiskit/algorithms/evolvers/variational/variational_principles/real_time_dependent_principle.py

```diff
@@ -34,27 +34,21 @@ class RealTimeDependentPrinciple(RealVariationalPrinciple):
     means that we consider real time dynamics.
     """
 
-    def calc_metric_tensor(
+    def create_qfi(
         self,
-        ansatz: QuantumCircuit,
-        parameters: List[Parameter],
-    ) -> ListOp:
+    ) -> QFI:
         """
-        Calculates a metric tensor according to the rules of this variational principle.
-
-        Args:
-            ansatz: Quantum state in the form of a parametrized quantum circuit to be used for
-                calculating a metric tensor.
-            parameters: Parameters with respect to which gradients should be computed.
+        Creates a QFI instance according to the rules of this variational principle. It is used
+        to calculate a metric tensor required in the ODE.
 
         Returns:
-            Transformed metric tensor.
+            QFI instance.
         """
-        raw_metric_tensor_imag = metric_tensor_calculator.calculate(
-            ansatz, parameters, self._qfi_method, basis=-Y
-        )
+        qfi_method = self._qfi_method
```

Comment:
This function could probably be avoided if we convert the `qfi_method` to a `CircuitQFI` with the correct settings in the initializer. That would be one less function to document and test 🙂